### PR TITLE
Merge payment frontend into backend

### PR DESCRIPTION
**Pull request by chuck-dbos**
The `ejs` code is now hosted in the backend; this simplifies deployment of payment significantly and shop overall somewhat.

There is no longer any need for the OpenAPI client for payment, so I took out the generator.  The generator lives on in shop-backend for use by shop-frontend, which is a sufficient illustration of how it works.

Slight changes to warning messages and comments in shop.

Stabilization of unit test - fix a timing hole.

Up SDK version to 0.11.8.  (Do that to other apps as well, apply audit fixes, to prevent the frustrations seen from affecting other apps.)

README section on cloud deployment.  (Ran through, and it worked.)

### CODE REVIEW

**Comment by manojdbos** [comment]
Have we tried deploying this in the cloud ?

**Comment by chuck-dbos** [comment]
> Have we tried deploying this in the cloud ?

We should... if it doesn't work is the app broken or is the cloud broken?

**Comment by manojdbos** [comment]
> > Have we tried deploying this in the cloud ?
> 
> We should... if it doesn't work is the app broken or is the cloud broken?

Let me know if you want me to try deploying this branch.

**Comment by manojdbos** [comment]
> > Let me know if you want me to try deploying this branch.
> 
> You can try a simple deploy if you like. I was going to do a cloud deployment and document it for the README at a later time... the README is very weak right now. But having one less server makes that a bit easier, so I wanted to do that part first. There has to be some dance to get the URLs for payment right. Then, I think the shop frontend would remain local, unless we figure out pushing it to vercel or similar.

I can try. We cannot merge until it works because cloud staging tests use shop and payment

**Comment by manojdbos** [comment]
> > I can try. We cannot merge until it works because cloud staging tests use shop and payment
> 
> Hmm. I would imagine as long as the staging test doesn't try to deploy payment frontend, it will work. Staging test having a tight dependency on `main` of shop demo is not so great as far as "chicken and egg" problems go, and it should be able to deploy a tagged version of the demo to break that. If shop demo changes, it won't break the test, once the test is updated it can pull a newer tag.

Tests deploy shop-backend and payment-backend. And I thought this PR combines payment-frontend and payment-backend

**Comment by manojdbos** [comment]
> > Tests deploy shop-backend and payment-backend. And I thought this PR combines payment-frontend and payment-backend
> 
> Yep it's still called `payment-backend` and ought to work, it just additionally serves a couple of `ejs` UI pages that it didn't before.

The staging shop test worked.
Going to try using the UI

**Comment by manojdbos** [comment]
> > > Tests deploy shop-backend and payment-backend. And I thought this PR combines payment-frontend and payment-backend
> > 
> > 
> > Yep it's still called `payment-backend` and ought to work, it just additionally serves a couple of `ejs` UI pages that it didn't before.
> 
> The staging shop test worked. Going to try using the UI

For the UI to work the payment URL sent back to shop needs to change 
from
https://mj.dev.dbos.dev/payment/e6d945b6-3276-4b05-a4c5-819d3b09f87a 
to
https://mj.dev.dbos.dev/apps/manoj/payment-backend/payment/ac0b3d97-c81f-4884-bcc4-b9d20849d472

File: e-commerce/payment-backend/dbos-config.yaml [hunk 1]
```diff
@@ -9,13 +9,13 @@ database:
...
-  user_dbclient: 'knex' # Will be deprecated, use app_db_client instead
   migrate: ['npx knex migrate:latest']
   rollback: ['npx knex migrate:rollback']
 runtimeConfig:
   port: 8086
 application:
-  frontend_host: 'http://localhost:8000'
+  frontend_host: 'http://localhost:8086'
+  # For DBOS Cloud:
+  #frontend_host: 'https://<username>-<app-name>.cloud.dbos.dev'
```
Comment (manojdbos) [inline e-commerce/payment-backend/dbos-config.yaml:21]: Should  this not be https://<username>-<app-name>.<clustername>.cloud.dbos.dev
May be not -- I guess there is no clustername for prod
So might be ok

Comment (chuck-dbos) [reply e-commerce/payment-backend/dbos-config.yaml:21]: One thing I could do is replace <app-name> with the name from package.json... that's unlikely to get changed. But it does sort of show the formula for the URL.

File: e-commerce/README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 This demo is a pair of [DBOS](https://github.com/dbos-inc/dbos-sdk) based systems demonstrating an 
 e-commerce scenario with separate apps for the online shop and the payment provider.
 
-## Demo Setup
+## Demo Setup (local)
```
Comment (kraftp) [inline e-commerce/README.md:6]: The install script still references the payment frontend

File: e-commerce/README.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
...
 
 This demo requires Node 20.x or later and a PostgreSQL compatible database.
 The demo includes a script to configure a PostgreSQL Docker container on your behalf.
```
Comment (kraftp) [inline e-commerce/README.md:9]: The docker script is for Postgres 15.4--should we update it to 16.1?

File: e-commerce/payment-backend/dbos-config.yaml (hunk 1 above)
Comment (maxdml) [reply e-commerce/payment-backend/dbos-config.yaml:21]: there is no cluster name for prod

**Comment by maxdml** [comment]
> shop-guide/package-lock.json
> shop-guide/package.json 

For these conflicts you can pick "theirs" @chuck-dbos  -- Peter removed some unused dependencies in shop-guide

File: e-commerce/payment-backend/dbos-config.yaml (hunk 1 above)
Comment (kraftp) [reply e-commerce/payment-backend/dbos-config.yaml:21]: It should be `https://<username>-payment-backend.cloud.dbos.dev`

File: e-commerce/README.md [hunk 4]
```diff
@@ -79,12 +72,71 @@ Add one or more writing utensils to your cart then press "Proceed to Checkout".
...
+* Register the application, using `npx dbos-cloud app register -d <dbname>`, with `<dbname>` set to match the name of the provisioned database server instance. (The payment app and shop app can share this database server instance, each will have its own database on that server instance.)
+* Adjust the `dbos-config.yaml` file, setting `payment_host` and `local_host` to your DBOS Cloud URLs.
+* Deploy the `shop-backend` app with `npx dbos-cloud app deploy`.
+
+### Shop Frontend
+The Shop Frontend is built in [Next.js](https://nextjs.org/) and does not (currently) deploy to DBOS Cloud.
+
+You can continue to run it locally:
+* Change to the shop frontend application directory (`e-commerce/shop-frontend`).
+* Set the environment variable `NEXT_PUBLIC_SHOP_BACKEND` to `https://<username>-<app-name>.cloud.dbos.dev/`
```
Comment (kraftp) [inline e-commerce/README.md:114]: `https://<username>-shop-backend.cloud.dbos.dev`

**Review by kraftp — APPROVED** [review]
Looks good, some minor fixes

File: e-commerce/README.md (hunk 3 above)
Comment (chuck-dbos) [reply e-commerce/README.md:9]: Will do that and add a PGPORT override.

**Review by manojdbos — APPROVED** [review]

**Pull request merged**